idf_build_apps/finder.py

```diff
@@ -29,7 +28,7 @@
 def _get_apps_from_path(
     path: str,
     target: str,
-    build_system: str = 'cmake',
+    app_cls: t.Type[App] = CMakeApp,
```

Comment:
better to keep the name

```suggestion
    build_system: t.Union[t.Type[App], str] = CMakeApp,
```

Reply:
I move `build_system -> app_cls` to method `find_apps`. 
It seems that this method `_get_apps_from_path` is an internal method. 
If I keep this method compatible with str parameters, I have to check the parameter type before the log in Line 156. Do you think it is important to let this method also be compatible with str parameter?

Reply:
yes you're right. I agree to keep as what it is right now